File: SDKs/amazonaws.com/opsworks/2013-02-18/python/src/sdk/models/shared/permission.py
import dataclasses
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils


@dataclass_json
@dataclasses.dataclass
class Permission:
    r"""Permission
    Describes stack or user permissions.
    """
    
    allow_ssh: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AllowSsh') }})
    allow_sudo: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('AllowSudo') }})
    iam_user_arn: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('IamUserArn') }})
    level: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('Level') }})
    stack_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('StackId') }})
    

File: SDKs/amazonaws.com/opsworks/2013-02-18/java/lib/src/main/java/openapisdk/models/shared/Permission.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * Permission
 * Describes stack or user permissions.
**/
public class Permission {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("AllowSsh")
    public Boolean allowSsh;
    public Permission withAllowSsh(Boolean allowSsh) {
        this.allowSsh = allowSsh;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("AllowSudo")
    public Boolean allowSudo;
    public Permission withAllowSudo(Boolean allowSudo) {
        this.allowSudo = allowSudo;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("IamUserArn")
    public String iamUserArn;
    public Permission withIamUserArn(String iamUserArn) {
        this.iamUserArn = iamUserArn;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("Level")
    public String level;
    public Permission withLevel(String level) {
        this.level = level;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("StackId")
    public String stackId;
    public Permission withStackId(String stackId) {
        this.stackId = stackId;
        return this;
    }
}